src/services/ConnectionService.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect, createContext, useRef } from "react";
+import { useDispatch } from "react-redux";
+import { Connection, createConnection } from "@models/Connection";
+import { updateConnection } from "@slices/connectionsSlice";
+
+export const ConnectionService = ({ children }: any) => {
+  const dispatch = useDispatch();
+  let connectionSocket = useRef<WebSocket>();
+  let connections: Connection[];
+
+  useEffect(() => {
+    connectionSocket.current = new WebSocket(
+      `ws://${import.meta.env.VITE_SERVER_IP}:${
+        import.meta.env.VITE_SERVER_PORT
+      }${import.meta.env.VITE_CONNECTIONS_URL}`
+    );
+
+    dispatch(updateConnection(createConnection("connections", false)));
```

Comment:
Change name of createConnection to setConnectionState as connection may already exists so you are not creating it.